pkg/scheduler/core/equivalence_cache.go

```diff
@@ -142,18 +147,14 @@ func (ec *EquivalenceCache) lookupResult(
 	if algorithmCache, exist := ec.algorithmCache[nodeName]; exist {
 		if cachePredicate, exist := algorithmCache.predicatesCache.Get(predicateKey); exist {
 			predicateMap := cachePredicate.(PredicateMap)
-			// TODO(resouer) Is it possible a race that cache failed to update immediately?
-			if hostPredicate, ok := predicateMap[equivalenceHash]; ok {
-				if hostPredicate.Fit {
-					return true, []algorithm.PredicateFailureReason{}, false
-				}
-				return false, hostPredicate.FailReasons, false
+			if hostPredicate, exist := predicateMap[equivalenceHash]; exist {
+				metrics.EquivalenceCacheHits.Inc()
+				return hostPredicate.Fit, hostPredicate.FailReasons, false
 			}
-			// is invalid
-			return false, []algorithm.PredicateFailureReason{}, true
 		}
 	}
-	return false, []algorithm.PredicateFailureReason{}, true
+	metrics.EquivalenceCacheMisses.Inc()
+	return false, nil, true
```

Comment:
Just curious why not return empty slice as before so we don't need to add a `slicesEqual()` later?

Reply:
The reason I switched from empty slice to nil is that it seemed like a shorter and cleaner way to return an "empty" slice in go.

The reason I "fixed" the test instead of reverting this line is that I don't feel the test was correct to assume that nil slice is invalid.

If you feel strongly about this, I can put it back.

Reply:
I am fine with the change, thanks for clarify.